tests/test-component.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Class Component_Tests.
+ *
+ * @package WP_Irving
+ */
+
+use WP_Irving\Component;
+
+/**
+ * Tests for the component class.
+ */
+class Component_Tests extends WP_UnitTestCase {
+
+	/**
+	 * Associative array of components to use for testing.
+	 *
+	 * @var array
+	 */
+	public static $components = [];
+
+	public function setUp() {
```

Comment:
`setUp()` will rerun the `create_test_components()` method before each test method. To set these up once and share them across methods, `wpSetUpBeforeClass()` would be a better choice.